libethereum/State.h

```diff
@@ -142,8 +143,9 @@ struct Change
 	{}
 };
 
-}
+typedef std::vector<Change> ChangeLog;
```

Comment:
1. `using ChangeLog = std::vector<Change>;
2. It must not be in `detail` namespace. Put it in `eth` namespace directly. Now the `detail` namespace is pointless as we now make the change log available outside of State.